src/Window.js

```diff
@@ -475,9 +476,18 @@ const _makeWindow = (options = {}, parent = null, top = null) => {
       if (nativeMl && nativeMl.IsPresent()) {
         result.push(window[symbols.mrDisplaysSymbol].mlDisplay);
       }
-      if (nativeVr && nativeVr.VR_IsHmdPresent()) {
-        result.push(window[symbols.mrDisplaysSymbol].vrDisplay);
+
+      // Oculus runtime takes precedence over OpenVR for Oculus headsets.
+      if (nativeOculusVR && nativeOculusVR.Oculus_IsHmdPresent()) {
+        GlobalContext.isOculusRuntime = true;
```

Comment:
I think thins might result in race conditions, where depending on whether you get the display or the controllers first, the controllers will be called something different. Additionally I don't think a global should be set from a getter.

Maybe the better approach would be to bootstrap controllers in this flow as well?

Reply:
I think if we follow spec / other browsers implementation the problem cannot happen because VR controllers only show up in the gamepad API after starting to present.

Reply:
I added a check to return empty array and not initialize gamepads if we're not presenting https://github.com/webmixedreality/exokit/pull/787/commits/600e9c254286e11f7edcd0ca534f35d213497a18

Reply:
> I think if we follow spec / other browsers implementation the problem cannot happen because VR controllers only show up in the gamepad API after starting to present.

Ok, that makes sense.

Is https://github.com/webmixedreality/exokit/commit/600e9c254286e11f7edcd0ca534f35d213497a18 the right commit?  I don't see any gamepads code there.

Reply:
In that case, should we move the check to the Gamepad getter entirely? If the intent is to return certain controllers when oculus is presenting, we can probably check whether the devices are actually presenting (https://github.com/webmixedreality/exokit/pull/787#diff-3db2392f75003eacb1df3b866c44fb3cR1362) rather than adding another flag to track it on display lookup time.